### ISSUE TITLE
Delete with data not working since v0.20.0

### ISSUE THREAD

**Issue by MBurchard** [axios/axios#3318]
#### Describe the bug
DELETE Request with data stopped working when installing v0.20.0.  
The data is not send any more.  
Switching back to v0.19.2 solved the problem.

#### To Reproduce
Code snippet to reproduce, ideally that will work by pasting into something like https://npm.runkit.com/axios, a hosted solution, or a repository that illustrates the issue. **If your problem is not reproducible, please file under Support or Usage Question**

```js
import axios from 'axios';

const http = axios.create({
	baseURL: '/api',
	timeout: 60000
});

const getAppHeader = (app) => {
	return {
		headers: {
			'X-ON-BEHALF-OF-APPLICATION': app.clientId
		}
	};
};

/**
 * Adds Users to the Application Manager Group for the given application.
 *
 * @param users
 * @param app
 * @return {Promise<boolean>}
 */
const changeAppMgrGroupMembers = async (app, users) => {
	const conf = getAppHeader(app);
	const usersRemove = [];
	const usersAdd = [];
	users.forEach((user) => {
		if (user.flagDelete) {
			usersRemove.push(user);
		} else {
			usersAdd.push(user);
		}
	});
	let result = true;
	try {
		const response = await http.delete('/AppMgrGroups', Object.assign({data: usersRemove}, conf));
		result = response.status === 202;
	} catch (e) {
		console.error('Error in addUsersToAppMgrGroup', e);
	}
	...
```

#### Expected behavior
Delete with data should continue working in v0.20. Much better would be, if it works same as post or push.

#### Environment
 - Axios Version [e.g. 0.20.0]
 - Adapter [e.g. XHR/HTTP]
 - Browser Firefox
 - Browser Version 81
 - Node.js Version 12.18.4
 - OS: Linux Mint 20
 - Additional Library Versions @vue/cli 4.5.6, vue 2

**Comment by jmsv** [comment]
Same issue here, looks like it's been fixed in #3282 but not released yet

**Comment by jasonsaayman** [comment]
Hi,

Version 0.21.0 has been released 🎉 please use that and let us know if that solves your issue.

Thanks
